giraffe/src/components/GeoLayer.tsx

```diff
@@ -1,5 +1,5 @@
 // Libraries
-import React, {FunctionComponent} from 'react'
+import React, {FunctionComponent, useState} from 'react'
```

Comment:
Suggestion: This component is kind of getting too big. Can we break this down into smaller components and move some of this function into a different files, For ex: `utils`?